### PR TITLE
commander_btc: Check the return value of snprintf.

### PR DESCRIPTION
**Pull request by conte91**
snprintf writing too much data would make the target
buffer contain a string with no terminating NULL byte. Check
that this is not happening before proceeding.

### CODE REVIEW

File: src/commander/commander_btc.c [hunk 1]
```diff
@@ -81,6 +82,13 @@ static commander_error_t _btc_pub_xpub(const BTCPubRequest* request, PubResponse
         default:
             return COMMANDER_ERR_GENERIC;
         }
+        if (n_written >= sizeof(title)) {
```
Comment (benma) [inline src/commander/commander_btc.c]: n_written < 0 ||

Comment (benma) [reply src/commander/commander_btc.c]: also move the whole thing to right after `n_written` is set. error handling should ideally be always the next line.

**Comment by benma** [comment]
Also pleae change the commit msg, `snprintf` does null terminate, even if it truncates.

File: src/commander/commander_btc.c (hunk 1 above)
Comment (conte91) [reply src/commander/commander_btc.c]: Fixed.

File: src/commander/commander_btc.c [hunk 2]
```diff
@@ -62,6 +62,7 @@ static commander_error_t _btc_pub_xpub(const BTCPubRequest* request, PubResponse
             return COMMANDER_ERR_GENERIC;
         }
         char title[100] = {0};
+        int n_written;
```
Comment (benma) [inline src/commander/commander_btc.c]: why not inline? principle of smallest scope.

Comment (conte91) [reply src/commander/commander_btc.c]: You can't allocate stack space within a `switch` statement!

**Pull request merged**